src/Provider/Mimmi20BrowserDetector.php

```diff
@@ -0,0 +1,241 @@
+use UaResult\Result\Result;
+use UserAgentParser\Exception\NoResultFoundException;
+use UserAgentParser\Exception\PackageNotLoadedException;
+use UserAgentParser\Model;
+
+/**
+ * Abstraction for mimmi20/browser-detector
+ *
+ * @author Martin Keckeis <martin.keckeis1@gmail.com>
+ * @license MIT
+ * @see https://github.com/donatj/PhpUserAgent
+ */
+class Mimmi20BrowserDetector extends AbstractProvider
+{
+    /**
+     * Name of the provider
+     *
+     * @var string
+     */
+    protected $name = 'Mimmi20BrowserDetector';
+
+    /**
+     * Homepage of the provider
+     *
+     * @var string
+     */
+    protected $homepage = 'https://github.com/mimmi20/BrowserDetector';
+
+    /**
+     * Composer package name
+     *
+     * @var string
+     */
+    protected $packageName = 'mimmi20/browser-detector';
+
+    protected $detectionCapabilities = [
+
+        'browser' => [
+            'name'    => true,
+            'version' => true,
+        ],
+
+        'renderingEngine' => [
+            'name'    => true,
+            'version' => false,
+        ],
+
+        'operatingSystem' => [
+            'name'    => true,
+            'version' => true,
+        ],
+
+        'device' => [
+            'model'    => true,
+            'brand'    => true,
+            'type'     => true,
+            'isMobile' => false,
+            'isTouch'  => true,
+        ],
+
+        'bot' => [
+            'isBot' => true,
+            'name'  => true,
+            'type'  => false,
+        ],
+    ];
+
+    protected $defaultValues = [
+
+        'general' => [
+            '/^unknown$/i',
+        ],
+    ];
+
+    /**
+     *
+     * @var Detector
+     */
+    private $parser;
+
+    /**
+     *
+     * @param  Detector                  $parser
+     * @throws PackageNotLoadedException
+     */
+    public function __construct(Detector $parser = null)
+    {
+        if ($parser === null) {
+            $this->checkIfInstalled();
+        }
+
+        $this->parser = $parser;
+    }
+
+    /**
+     *
+     * @return Detector
+     */
+    public function getParser()
+    {
+        if ($this->parser !== null) {
+            return $this->parser;
+        }
+
+        $cache  = new MemoryCacheItemPool();
+        $logger = new NullLogger();
+
+        $this->parser = new Detector($cache, $logger);
+
+        return $this->parser;
+    }
+
+    /**
+     *
+     * @param Result $result
+     *
+     * @return bool
+     */
+    private function hasResult(Result $result)
+    {
+        if ($this->isRealResult($result->getBrowser()
+            ->getType()
+            ->getType())) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     *
+     * @param Model\Bot        $bot
+     * @param BrowserInterface $browserRaw
+     */
+    private function hydrateBot(Model\Bot $bot, BrowserInterface $browserRaw)
+    {
+        $bot->setIsBot(true);
+        $bot->setName($this->getRealResult($browserRaw->getName()));
```

Comment:
add this: `$bot->setType($this->getRealResult($browserRaw->getType()->getName()));`